### PR TITLE
ilmbase, openexr@2: fix livecheck

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

IlmBase exists only for version 2.x.y. See #76325.

Closes #106863.

### CODE REVIEW

**Comment by carlocab** [comment]
Incidentally, this picks up a bugfix 2.5.8.

**Review by samford — COMMENTED** [review]
Without prior knowledge of this software, it strikes me that `openexr@2` is the parent rather than the child (considering the repository is named `openexr`). I would think this should be the other way around, with `openexr@2` containing the  `livecheck` block and `ilmbase` using `formula "openexr@2"`. Is there any particular reason behind the current setup in this PR?

For example, if we're concerned that one of these formulae will be deprecated before the other (where a `livecheck` block reference could point to a deprecated formula instead of a working check), then it may be better to simply duplicate this `livecheck` block.

**Comment by carlocab** [comment]
> Is there any particular reason behind the current setup in this PR?

`openexr@2` depends on `ilmbase`, so it seemed sensible for that to be the "base" formula. Though I'm
not really sure how this is important since `livecheck` behaves the same way in either case.

> For example, if we're concerned that one of these formulae will be deprecated before the other (where a `livecheck` block reference could point to a deprecated formula instead of a working check), then it may be better to simply duplicate this `livecheck` block.

Our deprecation policy requires that `ilmbase` cannot be deprecated unless `openexr@2` is also deprecated. That will be enforced through an audit after https://github.com/Homebrew/brew/pull/12770.

This means that it should never be the case that `ilmbase` is deprecated before `openexr@2`.

File: Formula/ilmbase.rb [hunk 1]
```diff
@@ -5,6 +5,11 @@ class Ilmbase < Formula
   sha256 "36ecb2290cba6fc92b2ec9357f8dc0e364b4f9a90d727bf9a57c84760695272d"
   license "BSD-3-Clause"
 
+  livecheck do
+    url :stable
+    regex(/^v?(2(?:\.\d+)+)$/i)
```
Comment (samford) [inline Formula/ilmbase.rb:10]:
```suggestion
    formula "openexr@2"
```

File: Formula/openexr@2.rb [hunk 2]
```diff
@@ -5,6 +5,10 @@ class OpenexrAT2 < Formula
   sha256 "36ecb2290cba6fc92b2ec9357f8dc0e364b4f9a90d727bf9a57c84760695272d"
   license "BSD-3-Clause"
 
+  livecheck do
+    formula "ilmbase"
```
Comment (samford) [inline Formula/openexr@2.rb:9]:
```suggestion
    url :stable
    regex(/^v?(2(?:\.\d+)+)$/i)
```

**Review by samford — COMMENTED** [review]
> `openexr@2` depends on `ilmbase`, so it seemed sensible for that to be the "base" formula. Though I'm not really sure how this is important since `livecheck` behaves the same way in either case.

Since we don't have to worry about deprecation order (thanks for confirming), it's simply a convention that when a package is part of an overarching project (as `ilmbase` is in the `openexr` repository), the main package is treated as the parent and ancillary packages are treated as children with respect to `livecheck` block references. There are sometimes practical reasons for this but it's only customary in this particular instance.

For example, `logcli`, `loki`, and `promtail` all share the same https://github.com/grafana/loki repository. `loki` is treated as the main formula and the other two contain `livecheck` blocks that reference it, as they're secondary to `loki` with respect to the repository. From my perspective, it's a similar situation here between `openexr@2` and `ilmbase`.

**Comment by carlocab** [comment]
We don't have a similar guarantee that `openexr@2` will not be deprecated before `ilmbase`, so it seems like your concern regarding the base formula getting deprecated first applies to your suggestion.

**Comment by samford** [comment]
Ah, I had misinterpreted something you said earlier. I'm also now realizing that `openexr@2` isn't the only formula depending on `ilmbase`.

I'm still kind of on the fence about this reverse reference, as it would be the only `livecheck` block reference (out of 61) that goes against the aforementioned pattern. That said, the only other option would be to duplicate the `livecheck` block. There isn't a functional difference either way and these `livecheck` blocks are unlikely to be modified before the formulae are deprecated, so I'll go ahead and approve this. Thanks for the discussion.

**Review by samford — APPROVED** [review]

**Pull request closed without merging**